### PR TITLE
set parallelism in integration tests using GOMAXPROCS

### PR DESCRIPTION
**Pull request by liggitt**
xref https://github.com/kubernetes/kubernetes/issues/109038#issuecomment-1085950690 and https://github.com/kubernetes/kubernetes/issues/109038#issuecomment-1086055164

There is evidence integration test runs are overloading the test machine with concurrent build/link/test runs

```release-note
NONE
```

### CODE REVIEW

**Comment by liggitt** [comment]
passed!

**Comment by liggitt** [comment]
we were already setting KUBE_INTEGRATION_TEST_MAX_CONCURRENCY to 4 in hack/jenkins/test-dockerized.sh

someone in the past learned this lesson, and some test refactors in the distant past stopped honoring this

File: hack/make-rules/test-integration.sh
Comment (aojea) [inline hack/make-rules/test-integration.sh:32]: ic, this is set in 

```
rep -r KUBE_INTEGRATION_TEST_MAX_CONCURRENCY hack/
hack/jenkins/test-dockerized.sh:export KUBE_INTEGRATION_TEST_MAX_CONCURRENCY=4
```

**Comment by aojea** [comment]
/kind flake
/prioritiy critical-urgent
/lgtm

I'm going to be AFK , in case you can get reviews I leave the lgtm and hold until you are satisfied with the tests

/hold

maybe is not only the `-p` and GOMAXPROCS is affecting something else ?

https://github.com/kubernetes/kubernetes/pull/109240

**Comment by liggitt** [comment]
2 out of 2 integration runs were green

File: hack/make-rules/test-integration.sh
Comment (BenTheElder) [reply hack/make-rules/test-integration.sh:32]: TODO: can we auto-set this better?
but at least we have a standard way to plumb it now.

**Review by BenTheElder — APPROVED** [review]
/lgtm

**Comment by BenTheElder** [comment]
/triage accepted
e2e flakes up too? :(

we used to have 90%+ passing, not so much today
https://prow.k8s.io/?job=pull-kubernetes-e2e-kind

**Comment by aojea** [comment]
OMG, didn't merge yet :eye:

**Comment by liggitt** [comment]
> OMG, didn't merge yet 👁️

at least integration is green 🤷

**Comment by BenTheElder** [comment]
We're all green 🤞🤞🤞

**Comment by liggitt** [comment]
batch with this in it hit a failure and is retrying - https://prow.k8s.io/?repo=kubernetes%2Fkubernetes&type=batch&pull=109239

if the batch fails, this should go in solo

**Comment by liggitt** [comment]
> e2e flakes up too? :(

yes :-(

**Comment by liggitt** [comment]
our short local nightmare has come to an end 🤞

**Pull request merged**
